uaa_client/authentication.py

```diff
@@ -93,13 +93,21 @@ def get_user_by_email(email):
         of security policies for logins.
         '''
 
+        APPROVED_DOMAINS = getattr(settings, 'UAA_APPROVED_DOMAINS', [])
+
         try:
             return User.objects.get(email__iexact=email)
         except User.DoesNotExist:
-            logger.info(
-                'User with email {} does not exist'.format(email)
-            )
-            return None
+            email_pieces = email.split('@')
+            if email_pieces[1] in APPROVED_DOMAINS:
+                User.objects.create_user(email_pieces[0], email)
```

Comment:
Hmm, we should probably document that we're using the username portion of their email address to set the username of the new user?

Hmm, I guess I'm a bit concerned about edge cases here...

According to the [Django docs](https://docs.djangoproject.com/en/1.11/ref/contrib/auth/#django.contrib.auth.models.User.username). "Usernames may contain alphanumeric, _, @, +, . and - characters". Should we be doing something along the lines of "slugifying" the username just in case the email doesn't work?

Also, what if there are username collisions for different emails? For example, a `john.doe@gsa.gov` and `john.doe@nasa.gov` might represent totally different people with the same name from different agencies, but this code might throw if the names collide. Blarg!

I guess one option might be to just start adding numbers to the username if this happens, so that the second `john.doe`'s username becomes `john.doe2` or something.